File: src/main/java/compiler/block/ifs/IfBlock.java
package compiler.block.ifs;

import compiler.Parameter;
import compiler.Utils;
import compiler.block.Block;
import compiler.symbol_table.SymbolTable;

public class IfBlock extends Block {

    private String type = "if";
    private Parameter[] params;
    String name;
    String pointer;
    String operator;
    String value;

    public IfBlock(Block superBlock, String name) {
        super(superBlock, true, false);
        this.name = name;

        String[] split = name.split(" ");

        //  x == 10
        if(split.length > 1) {
            pointer = split[0];
            pointer = ""+ SymbolTable.getInstance().getValue(Utils.getMethod(this), split[0]).getId();
            operator = split[1];
            value = split[2];
        }else{
            //boolean value
            value = name;

        }

    }

    public String getName() {
        return name;
    }

    @Override
    public String getValue() {
        return null;
    }

    public String getType() {
        return type;
    }

    @Override
    public String getOpeningCode() {
        return  "mv.visitVarInsn(ILOAD,"+pointer+");" +
                "mv.visitLdcInsn("+value+");\n" +
                "Label l"+id+" = new Label();\n" +
                "mv.visitJumpInsn(IF_ICMPNE, l"+id+");";
    }

    @Override
    public String getBodyCode() {
        return "";
    }

    @Override
    public String getClosingCode() {
        return "mv.visitLabel(l"+id+");"
               ;
    }



    public Parameter[] getParameters() {
        return params;
    }


    @Override
    public void run() {

    }
}

File: src/main/scala/compiler/block/ifs/IfBlock.scala
package compiler.block.ifs

import compiler.block.Block
import compiler.generators.ifs.IfGen
import compiler.structure.parameters.Parameter
import compiler.symbol_table.SymbolTable
import compiler.utilities.Utils

/**
  * Represents an if statement
  *
  * @param superBlockInit
  * @param nameInit
  */
class IfBlock(var superBlockInit: Block, var nameInit: String) extends Block(superBlockInit, true, false) {

  private val params: Array[Parameter] = null
  private val _name = nameInit
  private val split: Array[String] = _name.split(" ")
  private val pointer: String = "" + SymbolTable.getInstance.getValue(Utils.getMethod(this), split(0)).getId
  private val operator: String = if(split.length > 1) split(1) else ""
  private val value: String = if(split.length > 1) split(2) else _name

  private val byteCodeOp: String = {
    if (operator == "==") {
      "mv.visitJumpInsn(IF_ICMPGE, l" + id + ");\n"
    }
    else if (operator == "<") {
      "mv.visitJumpInsn(IF_ICMPGE, l" + id + ");\n"
    }
    else if (operator == ">") {
      "mv.visitJumpInsn(IF_ICMPGE, l" + id + ");\n"
    }
    else if (operator == "<=") {
      "mv.visitJumpInsn(IF_ICMPGE, l" + id + ");\n"
    }
    else if (operator == ">=") {
      "mv.visitJumpInsn(IF_ICMPGE, l" + id + ");\n"
    }
    else {
      throw new RuntimeException("Error: Disallowed Operator")
    }
  }

  def getParameters: Array[Parameter] = params

  def getName: String = _name

  def getType: String = "if"

  def getValue: String = null

  def init() {
  }

  def getOpeningCode: String = {
    return IfGen.getOpeningCode(pointer, value, id, byteCodeOp)
  }

  def getClosingCode: String = {
    return asm.visitLabel("l" + id)
  }

  override def toString: String = "if " + _name

}